server/ai_mediaserver.go

```diff
@@ -657,47 +657,65 @@ func processStream(ctx context.Context, params aiRequestParams, req worker.GenLi
 	go func() {
 		var err error
 		for {
-			perOrchCtx, perOrchCancel := context.WithCancel(ctx)
+			perOrchCtx, perOrchCancel := context.WithCancelCause(ctx)
 			params.liveParams = newParams(params.liveParams, perOrchCancel)
 			var resp interface{}
 			resp, err = processAIRequest(perOrchCtx, params, req)
 			if err != nil {
 				clog.Errorf(ctx, "Error processing AI Request: %s", err)
-				perOrchCancel()
+				perOrchCancel(err)
 				break
 			}
 
 			if err = startProcessing(perOrchCtx, params, resp); err != nil {
 				clog.Errorf(ctx, "Error starting processing: %s", err)
-				perOrchCancel()
+				perOrchCancel(err)
 				break
 			}
 			if isFirst {
 				isFirst = false
 				firstProcessed <- struct{}{}
 			}
 			<-perOrchCtx.Done()
+			err = context.Cause(perOrchCtx)
+			if errors.Is(err, context.Canceled) {
+				// this happens if parent ctx was cancelled without a CancelCause
+				// or if passing `nil` as a CancelCause
+				err = nil
+			}
 			if !params.inputStreamExists() {
 				clog.Info(ctx, "No input stream, skipping orchestrator swap")
 				break
 			}
 			if !orchSwapper.shouldSwap(ctx) {
-				err = errors.New("Not swapping: kicking")
+				// TODO return an error from shouldSwap
```

Comment:
Why not just add it in this PR?

Reply:
Mostly trying to contain the scope of changes in this PR, but I can add it.